src/transaction.rs

```diff
@@ -85,7 +114,9 @@ impl Transaction {
         self.commit_digest = self.commit_digest.dot(&statement_hash);
 
         let mut values = vec![];
-        let mut current = first_page;
+        let mut current = execute_result.first_page;
+        let mut cumulative_timing = execute_result.timing_information.clone();
```

Comment:
These don't mutate do they? My Rust-fu is probably off but do the handles change?

Reply:
They don't. https://doc.rust-lang.org/book/ch03-01-variables-and-mutability.html explains it better than I can!